swarm/storage/mru/resource.go

```diff
@@ -274,47 +235,43 @@ func NewHandler(params *HandlerParams) (*Handler, error) {
 	return rh, nil
 }
 
-// SetStore sets the store backend for resource updates
+// SetStore sets the store backend for the Mutable Resource API
 func (h *Handler) SetStore(store *storage.NetStore) {
 	h.chunkStore = store
 }
 
-// Validate is a chunk validation method (matches ChunkValidatorFunc signature)
-//
-// If resource update, owner is checked against ENS record of resource name inferred from chunk data
-// If parsed signature is nil, validates automatically
-// If not resource update, it validates are root chunk if length is metadataChunkOffsetSize and first two bytes are 0
+// Validate is a chunk validation method
+// If it's a resource update, the chunk address is checked against the public key of the update's signature
+// It implements the storage.ChunkValidator interface
 func (h *Handler) Validate(addr storage.Address, data []byte) bool {
+
+	// does the chunk data make sense?
+	// This is not 100% safe evaluation, since content addressed data can coincidentally produce data with length header matching content size
 	signature, period, version, name, parseddata, _, err := h.parseUpdate(data)
 	if err != nil {
-		log.Warn(err.Error())
-		if len(data) > metadataChunkOffsetSize { // identifier comes after this byte range, and must be at least one byte
-			if bytes.Equal(data[:2], []byte{0, 0}) {
-				return true
-			}
-		}
-		log.Error("Invalid resource chunk")
+		log.Warn("Invalid resource chunk")
```

Comment:
It would be very helpful to have `addr` value in the log message.

Reply:
@jpeletier you put this too in your future PR?

Reply:
I'll do that. Thanks :)